teensy4-bsp/src/usb.rs

```diff
@@ -188,7 +188,7 @@ impl<'a> AsRef<[u8]> for Cursor<'a> {
 }
 
 /// A type that can read USB serial messages from a host
-pub struct Reader(core::marker::PhantomData<*const ()>);
+pub struct Reader(());
```

Comment:
Although they're both zero-sized types, the one that wraps `PhantomData<*const ()>` is neither `Send` nor `Sync`. But, the one that wraps `()` is both `Send` and `Sync`. We demonstrate those behaviors [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=45fd4f52c6625c478fe1a1a0d33ff985).

This change will let users access and modify a `Reader` across execution contexts. I'm not sure that's appropriate right now, given that the underlying implementation modifies static, shared state without synchronization. I'm concerned this could put the USB stack into an inconsistent state if, for example, a user is reading from the USB endpoint in both `main()` and a timer ISR.

I'm also not sure if we should commit to these trait bounds. Suppose that we re-write the USB stack in Rust. The `Reader` is more complex than a ZST, and it cannot be one of `Send` or `Sync`. The re-write could break our user's code if they were previously relying on `Reader: Send` or `Reader: Sync`.

I'd rather be conservative and say it's neither `Send` nor `Sync`, since adding these capabilities in the future if backwards-compatible. Given these considerations, is there a strong reason to consider `()` over `PhantomData<*const ()>` as the inner type of `Reader`?

Reply:
I see, I didn't know the motivation behind this, I'll change it back! (might help to copy-paste your PR comment into the code?)